hazelcast/src/main/java/com/hazelcast/mapreduce/impl/task/DefaultContext.java

```diff
@@ -102,11 +102,12 @@ public void finalizeCombiners() {
         Combiner<ValueIn, ?> combiner = combiners.get(key);
         if (combiner == null) {
             combiner = combinerFactory.newCombiner(key);
```

Comment:
It will work, but computeIfAbsent would let to use it in another places.

Actually I think that if we can write a code without lock it is better to write it without lock.